stripe/resource_stripe_meter.go

```diff
@@ -0,0 +1,299 @@
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The time window to pre-aggregate meter events for, if any.",
+			},
+			"value_settings": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				ForceNew:    true,
+				MaxItems:    1,
+				Description: "Fields that specify how to calculate a meter event’s value.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"event_payload_key": {
+							Type:     schema.TypeString,
+							Required: true,
+							Description: `The key in the usage event payload to use as the ` +
+								`value for this meter. For example, if the event payload  ` +
+								`contains usage on a bytes_used field, then set the ` +
+								`event_payload_key to “bytes_used”`,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceStripeMeterRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.API)
+	var meter *stripe.BillingMeter
+	var err error
+
+	err = retryWithBackOff(func() error {
+		params := &stripe.BillingMeterParams{}
+
+		meter, err = c.BillingMeters.Get(d.Id(), params)
+		return err
+	})
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return CallSet(
+		d.Set("display_name", meter.DisplayName),
+		d.Set("event_name", meter.EventName),
+		d.Set("event_time_window", meter.EventTimeWindow),
+		func() error {
+			if meter.DefaultAggregation != nil {
+				return d.Set("default_aggregation", []map[string]interface{}{
+					{
+						"formula": meter.DefaultAggregation.Formula,
+					},
+				})
+			}
+			return nil
+		}(),
+		func() error {
+			if meter.CustomerMapping != nil {
+				return d.Set("customer_mapping", []map[string]interface{}{
+					{
+						"event_payload_key": meter.CustomerMapping.EventPayloadKey,
+						"type":              meter.CustomerMapping.Type,
+					},
+				})
+			}
+			return nil
+		}(),
+		func() error {
+			if meter.ValueSettings != nil {
+				return d.Set("value_settings", []map[string]interface{}{
+					{
+						"event_payload_key": meter.ValueSettings.EventPayloadKey,
+					},
+				})
+			}
+			return nil
+		}(),
+	)
+}
+
+func resourceStripeMeterCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.API)
+	var meter *stripe.BillingMeter
+	var err error
+
+	params := &stripe.BillingMeterParams{
+		DisplayName:   stripe.String(ExtractString(d, "display_name")),
+		EventName:     stripe.String(ExtractString(d, "event_name")),
+		ValueSettings: &stripe.BillingMeterValueSettingsParams{},
+	}
+
+	if eventTimeWindow, set := d.GetOk("event_time_window"); set {
+		params.EventTimeWindow = stripe.String(ToString(eventTimeWindow))
+	}
+
+	if defaultAggregation, set := d.GetOk("default_aggregation"); set {
+		params.DefaultAggregation = &stripe.BillingMeterDefaultAggregationParams{}
+		defaultAggregationMap := ToMap(defaultAggregation)
+		for k, v := range defaultAggregationMap {
+			switch {
+			case k == "formula" && ToString(v) != "":
```

Comment:
I think we don't have to check for empty string `ToString(v) != ""` when both ends in error - either missing value in the call to Stripe API or error from the API.

I would rather not check for it because Stripe gives you more specific error.